x-pack/plugins/cases/server/common/utils.ts

```diff
@@ -258,7 +259,7 @@ export const isCommentRequestTypeAlert = (
  * A type narrowing function for external reference so attachments.
  */
 export const isCommentRequestTypeExternalReferenceSO = (
-  context: Partial<CommentRequest>
+  context: Partial<AttachmentRequestAttributes>
```

Comment:
Shouldn't we do the same for the other utility functions? Even if we don't get TS errors now it will protects us from mistakes in the future.